src/main/java/picard/illumina/parser/IlluminaDataProviderFactory.java

```diff
@@ -256,45 +223,75 @@ public void setApplyEamssFiltering(final boolean applyEamssFiltering) {
      */
-    public BaseIlluminaDataProvider makeDataProvider(List<Integer> requestedTiles) {
+    public BaseIlluminaDataProvider makeDataProvider(List<Integer> requestedTiles, File barcodeFile) {
         if (requestedTiles == null) {
             requestedTiles = availableTiles;
         } else {
             if (requestedTiles.isEmpty()) {
                 throw new PicardException("Zero length tile list supplied to makeDataProvider, you must specify at least 1 tile OR pass NULL to use all available tiles");
             }
         }
+        if (IlluminaFileUtil.hasCbcls(basecallDirectory, lane)) {
+            final File laneDir = new File(basecallDirectory, IlluminaFileUtil.longLaneStr(lane));
 
-        final Map<IlluminaParser, Set<IlluminaDataType>> parsersToDataType = new HashMap<>();
-        for (final Map.Entry<SupportedIlluminaFormat, Set<IlluminaDataType>> fmToDt : formatToDataTypes.entrySet()) {
-            parsersToDataType.put(makeParser(fmToDt.getKey(), requestedTiles), fmToDt.getValue());
-        }
+            final File[] cycleDirs = IOUtil.getFilesMatchingRegexp(laneDir, IlluminaFileUtil.CYCLE_SUBDIRECTORY_PATTERN);
 
-        log.debug("The following parsers will be used by IlluminaDataProvider: " + StringUtil.join("," + parsersToDataType.keySet()));
+            //CBCLs
+            final List<File> cbcls = Arrays.stream(cycleDirs)
+                    .flatMap(cycleDir -> Arrays.stream(IOUtil.getFilesMatchingRegexp(cycleDir,
+                            "^" + IlluminaFileUtil.longLaneStr(lane) + "_(\\d{1,5}).cbcl$"))).collect(Collectors.toList());
 
-        return new IlluminaDataProvider(outputMapping, parsersToDataType, basecallDirectory, lane);
+            if (cbcls.size() == 0) {
+                throw new PicardException("No CBCL files found.");
+            }
+
+            IOUtil.assertFilesAreReadable(cbcls);
+
+            //locs
+            final List<AbstractIlluminaPositionFileReader.PositionInfo> locs = new ArrayList<>();
+            final File locsFile = new File(basecallDirectory.getParentFile(), AbstractIlluminaPositionFileReader.S_LOCS_FILE);
+            IOUtil.assertFileIsReadable(locsFile);
+            try (LocsFileReader locsFileReader = new LocsFileReader(locsFile)) {
+                while (locsFileReader.hasNext()) {
+                    locs.add(locsFileReader.next());
+                }
+            }
+
+            //filter
+            final Pattern laneTileRegex = Pattern.compile(ParameterizedFileUtil.escapePeriods(
+                    ParameterizedFileUtil.makeLaneTileRegex(".filter", lane)));
+            final File[] filterFiles = getTiledFiles(laneDir, laneTileRegex);
+
+            IOUtil.assertFilesAreReadable(Arrays.asList(filterFiles));
+            //UGH new provider does not support multiple tiles at a time.
+            return new NewIlluminaDataProvider(cbcls, locs, filterFiles, lane, requestedTiles.get(0), outputMapping, barcodeFile);
```

Comment:
So I think we need to fix this.  The options I see are:
1. Narrow the API to only allow requesting a single tile at a time and fix anywhere that does anything else
2. Fix NewIlluminaDataProvider to work for multiple tiles
3. Add a `MultiTileIlluminaDataProvider` that can wrap an existing DataProvider/Factory and add multi-tile support.

Thoughts?

Reply:
I went with #2. It seemed like the most straightforward change, but I could be convinced otherwise if you think the other methods have merit. The only place in the code base that actually uses a multiple tile provider is ExtractIlluminaBarcodes, so doing #1 could likely be easier, but this seems the most flexible.